contracts/EscrowFactoryWithERC20.sol

```diff
@@ -0,0 +1,51 @@
+
+    function createErc20Escrow(
+        address tknAddr,
+        uint escrowAmount,
+        uint timelock,
+        address payable escrowerReserve,
+        address escrowerTrade,
+        address escrowerRefund,
+        address payable payeeReserve,
+        address payeeTrade
+    )
+    public
+    {
+        bytes32 escrowParamsHash = keccak256(abi.encodePacked(
+            address(this),
```

Comment:
`address(this)` isn't necessary here because it's already factored into the address for the create2 deployment. It also doesn't hurt though.
create2 address calculation: `keccak256( 0xff ++ address ++ salt ++ keccak256(init_code))`

Reply:
Same with the `escrowParamsHash` calculation in `EscrowFactory`

Reply:
Yes thats true its not necessary as part of the salt for create2. Although we still emit this value as an indexed parameter in the `EscrowCreated` event so its nice to make sure this value is guaranteed to be unique even across different escrow factory deployments.